x/feegrant/types/filtered_fee.go

```diff
@@ -0,0 +1,94 @@
+package types
+
+import (
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/codec/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	proto "github.com/gogo/protobuf/proto"
+)
+
+var _ FeeAllowanceI = (*FilteredFeeAllowance)(nil)
+var _ types.UnpackInterfacesMessage = (*FilteredFeeAllowance)(nil)
+
+// UnpackInterfaces implements UnpackInterfacesMessage.UnpackInterfaces
+func (a *FilteredFeeAllowance) UnpackInterfaces(unpacker types.AnyUnpacker) error {
+	var allowance FeeAllowanceI
+	return unpacker.UnpackAny(a.Allowance, &allowance)
+}
+
+// NewFilteredFeeAllowance creates new filtered fee allowance.
+func NewFilteredFeeAllowance(allowance FeeAllowanceI, allowedMsgs []string) (*FilteredFeeAllowance, error) {
+	msg, ok := allowance.(proto.Message)
+	if !ok {
+		return nil, sdkerrors.Wrapf(sdkerrors.ErrPackAny, "cannot proto marshal %T", msg)
+	}
+	any, err := types.NewAnyWithValue(msg)
+	if err != nil {
+		return nil, err
+	}
+
+	return &FilteredFeeAllowance{
+		Allowance:       any,
+		AllowedMessages: allowedMsgs,
+	}, nil
+}
+
+// GetAllowance returns allowed fee allowance.
+func (a *FilteredFeeAllowance) GetAllowance() FeeAllowanceI {
+	allowance, ok := a.Allowance.GetCachedValue().(FeeAllowanceI)
+	if !ok {
+		return nil
```

Comment:
shouldn't we return an error here? we do `a.GetAllowance().Accept()` below, which might throw

Note: 0.43 hasn't been released yet, there's still time to add an error return value.

Reply:
> shouldn't we return an error here? we do `a.GetAllowance().Accept()` below, which might throw

I think it is not needed @amaurym. we are not calling `Accept` method inside `GetAllowance`.

Reply:
>I think it is not needed

Is it not needed because L42 is dead code (never called), or because returning `nil` is okay? I believe the latter is not true, because of L54 `a.GetAllowance().Accept()` will throw.

If L42 is dead code, maybe we should panic instead?

Anyways returning an error seems like the safe & correct bet here (btw same for `func (msg MsgGrantFeeAllowance) GetFeeAllowanceI()`).

Reply:
In L:42 we should definitely return an `sdkerrors.ErrLogic` and have a good log somewhere.

Reply:
> > I think it is not needed
> 
> Is it not needed because L42 is dead code (never called), or because returning `nil` is okay? I believe the latter is not true, because of L54 `a.GetAllowance().Accept()` will throw.
> 
> If L42 is dead code, maybe we should panic instead?
> 
> Anyways returning an error seems like the safe & correct bet here (btw same for `func (msg MsgGrantFeeAllowance) GetFeeAllowanceI()`).

Yeah, got it. thanks!